source/server/server.cc

```diff
@@ -445,6 +445,23 @@ void InstanceImpl::initialize(Network::Address::InstanceConstSharedPtr local_add
               absl::StrJoin(info.registered_headers_, ","));
   }
 
+  // Initialize the regex engine and inject to singleton.
+  // Needs to happen before stats store initialization because the stats
+  // matcher config can include regexes.
+  if (bootstrap_.has_default_regex_engine()) {
+    const auto& default_regex_engine = bootstrap_.default_regex_engine();
+    Regex::EngineFactory& factory =
+        Config::Utility::getAndCheckFactory<Regex::EngineFactory>(default_regex_engine);
+    auto config = Config::Utility::translateAnyToFactoryConfig(
```

Comment:
IIRC this may increase a stats counter ([deprecated_feature_use](https://www.envoyproxy.io/docs/envoy/latest/configuration/operations/runtime#runtime-stats)).
cc @mattklein123 can this code be performed before stats_store_ is fully initialized?

Reply:
I'm not sure we have a choice. IIRC there is code that handles this but I can't remember. I.e. I thought at some point I fixed it to reparse bootstrap to look for deprecated fields but maybe I'm not remembering.

Reply:
This block of code won't increment the stat, but the instantiation of the stats matcher in the block below this could if there's a regex matcher with a deprecated `google_re2: {}` config block. This doesn't block server startup, but the deprecated feature stat won't get incremented in this case.